private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -791,6 +793,83 @@ export class TemplateServiceClient {
     return { success: true, result: resultTemplates };
   }
 
+  /**
+   * @public
+   * Get entry point.
+   * Returns the specified template with the data bound into the template JSON.
+   * If a version is passed in, it will update the template JSON of that version.
+   * If not version is passed, it will update the latest version.
+   * @param {string} templateId - unique template id
+   * @param {JSON} data - data json to bound with template json
+   * @param {string} version - version number, used with templateId
+   * @param {boolean} isClient - used to ignore updating the hit number on a template
+   * @returns Promise as valid json
+   */
+  public async bindData(
+    templateId: string,
+    data: JSON,
+    version?: string,
+    isClient?: boolean,
+    token?: string,
+
+
+  ): Promise<JSONResponse<ITemplate[]>> {
+    let authCheck = this._checkAuthenticated(token);
+    if (!authCheck.success) {
+      return authCheck;
+    }
+    let authId = this.authProvider.getAuthIDFromToken(token || this.authProvider.token);
+    let userResponse = await this._getUser(authId);
+    if (!userResponse.success || !userResponse.result || userResponse.result.length === 0) {
+      return { success: false, errorMessage: userResponse.errorMessage };
+    }
+
+    let userId = userResponse.result![0]._id;
+    const templateQuery: Partial<ITemplate> = {
+      _id: templateId,
+    };
+
+    let response = await this.storageProvider.getTemplates(templateQuery, undefined, undefined);
+
+    if (!response.success || !response.result) return response;
+
+    let templates: ITemplate[] = response.result;
+
+    if (templates.length != 1) {
+      return { success: false, errorMessage: "Invalid template ID" };
+    }
+
+    if (isClient === undefined || isClient === false) {
+      // Update hit counter for template
+      this._incrementTemplateHits(templateId, templates![0], version);
+    }
+
+    sortTemplateByVersion(templates![0]);
+
+    let resultTemplates: ITemplate[] = [];
+    let template = templates[0];
+    if (template.isLive === false && template.owner !== userId) return { success: false, errorMessage: "Invalid user ID" };
+    if (!template.instances) { return { success: false, errorMessage: "Invalid template ID" } };
+    if (version) {
+      for (let instance of template.instances) {
```

Comment:
There's a bit of repetition with the creating card, editing json and pushing the template instance - can maybe edit to something like this? 
```
  // Default latest version
    let selectedInstance = template.instances![0];
    if (!version) {
      for (let instance of template.instances) {
        if (instance.version === version) {
          selectedInstance = instance;
          break;
        }
      }
    }
    let boundJSON: JSON = createCard(selectedInstance.json, data);
    selectedInstance.json = boundJSON;
    template.instances = [selectedInstance];
    return { success: true, result: [template] }  
```